packages/components/src/utils/getTextAsHtml.ts

```diff
@@ -20,31 +22,92 @@ export const getTextAsHtml = (content?: (HardBreakProps | TextProps)[]) => {
     return ""
   }
 
-  return content
-    .map((node) => {
-      if (node.type === "hardBreak") {
-        return "<br />"
-      }
+  const output: string[] = []
+  let previousNodeLinkMark: Marks | undefined = undefined
 
-      if (!node.marks) {
-        return node.text
+  // At every step, we will close off all marks except for links
+  // First encounter with a link, always open it first before other marks
+  // Close all other marks first before closing the link mark
+  content.forEach((node) => {
+    if (node.type === "hardBreak") {
+      // Close off the existing link mark if it exists
+      if (previousNodeLinkMark) {
+        output.push(`</${MARK_DOM_MAPPING.link}>`)
+        previousNodeLinkMark = undefined
       }
 
-      let output = node.text
+      output.push("<br />")
+      return
+    }
+
+    const currentNodeLinkMark = node.marks?.find((mark) => mark.type === "link")
+    const isLinkMarkNew =
+      !!currentNodeLinkMark &&
+      !isEqual(previousNodeLinkMark, currentNodeLinkMark)
+
+    // Close off the existing link mark if it is different
+    if (isLinkMarkNew && !!previousNodeLinkMark) {
+      output.push(`</${MARK_DOM_MAPPING.link}>`)
+      previousNodeLinkMark = undefined
+    }
+
+    // If there are no marks, just push the text
+    if (!node.marks) {
+      output.push(node.text)
+      return
+    }
+
+    if (isLinkMarkNew) {
+      previousNodeLinkMark = currentNodeLinkMark
+
+      // Sort such that the link mark is the first item
+      node.marks.sort((a, b) => {
+        if (a.type === "link") {
+          return -1
+        }
+
+        return 1
+      })
+
       node.marks.forEach((mark) => {
         if (mark.type === "link") {
-          output = `<${MARK_DOM_MAPPING[mark.type]} href="${
-            mark.attrs.href
-          }">${output}</${MARK_DOM_MAPPING[mark.type]}>`
-          return
+          output.push(
+            `<${MARK_DOM_MAPPING.link} target="${mark.attrs.target || "_self"}" href="${mark.attrs.href}">`,
+          )
+        } else {
+          output.push(`<${MARK_DOM_MAPPING[mark.type]}>`)
         }
-
-        output = `<${MARK_DOM_MAPPING[mark.type]}>${output}</${
-          MARK_DOM_MAPPING[mark.type]
-        }>`
       })
+    } else {
+      // Continue with the rest of the marks
+      node.marks
+        .filter((mark) => mark.type !== "link")
+        .forEach((mark) => {
+          output.push(`<${MARK_DOM_MAPPING[mark.type]}>`)
+        })
+    }
+
+    // Push the text
+    output.push(node.text)
+
+    // Close off all marks except for links in reverse order
+    const marksToClose = node.marks.filter((mark) => mark.type !== "link")
+    while (marksToClose.length) {
+      const mark = marksToClose.pop()
```

Comment:
just fyi, this pops off `node.marks` which might potentially be problematic if we are relying on it downstream or what not. 

could we be abit more paranoid and do a full clone here?

Reply:
Hmm in this case a shallow clone was already performed in line 94, so the pop here would not have any effect on the original `node.marks` array.